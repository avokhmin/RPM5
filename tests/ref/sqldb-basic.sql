-- Configure the SQL parser parameters to known defaults.
.echo on
.explain off
.headers on
-- .log stderr
.mode list
.nullvalue ""
.output stdout
.prompt 'foo> ' '---> '
.separator "|"
.timeout 1000
.timer off
.show

-- Instantiate the rpmdb.sql schema.
.read ref/rpmdb.sql

-- Basic rpmdbAdd/rpmdbRemove operations using SQL triggers.
BEGIN TRANSACTION;
INSERT into Packages (h) VALUES ('bing-1.2-3.noarch');
INSERT into Packages (h) VALUES ('bang-4.5-6.noarch');
INSERT into Packages (h) VALUES ('boom-7.8.9.noarch');
COMMIT TRANSACTION;

SELECT * from Packages;
SELECT * from Nvra;

BEGIN TRANSACTION;
DELETE FROM Packages WHERE i = 2;
COMMIT TRANSACTION;

SELECT * from Packages;
SELECT * from Nvra;

-- Basic .foo SQL metadata tests.
.backup main tmp/main.bak
.restore main tmp/main.bak
.dump main

.databases
.indices Packages
.schema
.tables

.exit
