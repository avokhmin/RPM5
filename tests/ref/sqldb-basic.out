-- Loading resources from /home/jbj/.sqliterc
.show
     echo: on
  explain: off
  headers: off
     mode: list
nullvalue: ""
   output: stdout
separator: "|"
    width: 
.echo on
.explain off
.headers on
.mode list
.nullvalue ""
.output stdout
.prompt 'foo> ' '---> '
.separator "|"
.timeout 1000
.timer off
.show
     echo: on
  explain: off
  headers: on
     mode: list
nullvalue: ""
   output: stdout
separator: "|"
    width: 
.read ref/rpmdb.sql
SELECT * from Packages;
i|h
1|bing-1.2-3.noarch
2|bang-4.5-6.noarch
3|boom-7.8.9.noarch
SELECT * from Nvra;
k|v
bing-1.2-3.noarch|1
bang-4.5-6.noarch|2
boom-7.8.9.noarch|3
SELECT * from Packages;
i|h
1|bing-1.2-3.noarch
3|boom-7.8.9.noarch
SELECT * from Nvra;
k|v
bing-1.2-3.noarch|1
boom-7.8.9.noarch|3
.backup main tmp/main.bak
.restore main tmp/main.bak
.dump main
PRAGMA foreign_keys=OFF;
BEGIN TRANSACTION;
COMMIT;
.databases
seq  name             file                                                      
---  ---------------  ----------------------------------------------------------
0    main             /X/src/wdj/tests/tmp/sqldb                                
1    temp                                                                       
.indices Packages
sqlite_autoindex_Packages_1
.schema
CREATE TABLE Nvra (
  k	TEXT PRIMARY KEY NOT NULL,
  v	INTEGER REFERENCES Packages(i) ON UPDATE RESTRICT ON DELETE RESTRICT
);
CREATE TABLE Packages (
  i	INTEGER UNIQUE PRIMARY KEY NOT NULL,
  h	BLOB NOT NULL
);
CREATE TRIGGER delete_Packages BEFORE DELETE ON Packages
  BEGIN
    DELETE FROM Nvra WHERE v = old.rowid;
  END;
CREATE TRIGGER insert_Packages AFTER INSERT ON Packages
  BEGIN
    INSERT INTO Nvra (k,v)	VALUES ( new.h, new.rowid );
  END;
.tables
Nvra      Packages
.exit
